src/hooks/useResizeObserver/useResizeObserver.ts

```diff
@@ -1,24 +1,37 @@
-import { type RefObject, useEffect } from 'react';
+import { useEffect } from 'react';
+import { unref, type Unreffable } from '../../index.js';
+import { useRefValue } from '../useRefValue/useRefValue.js';
 
 /**
- * This hook allows you to add a ResizeObserver for an element and remove it
+ * This hook allows you to add a ResizeObserver to an element and remove it
  * when the component unmounts.
  *
  * @param ref - The ref to observe
  * @param callback - The callback to fire when the element resizes
  */
-export function useResizeObserver(ref: RefObject<Element>, callback: ResizeObserverCallback): void {
+export function useResizeObserver(
```

Comment:
Do we want a separate ticket to support `unobserve`, e.g. having a boolean arg that defaults to `true`, but can be set to `false` to unobserve ?

Reply:
I think this behaviour is very unlikely and I don't know if I like a 4th parameter for the hook function signature.

```ts
function useResizeObserver(
  target: Unreffable<Element>,
  callback: ResizeObserverCallback,
  options?: ResizeObserverOptions,
  enabled = true,
): void
```

Implementation will be similar to this:

```ts
export function useResizeObserver(
  target: Unreffable<Element>,
  callback: ResizeObserverCallback,
  options?: ResizeObserverOptions,
  enabled = true,
): void {
  const memoizedOptions = useMemo(
    () => options,
    [JSON.stringify(options)],
  );

  const callbackRef = useRefValue(callback);
  const resizeObserver = useClientSideValue(
    () =>
      new ResizeObserver((entries, observer) => {
        callbackRef.current?.(entries, observer);
      }),
  );

  useEffect(() => {
    const element = unref(target);

    if (element === null) {
      return;
    }

    if (enabled) {
      resizeObserver?.observe(element, memoizedOptions);
    } else {
      resizeObserver?.unobserve(element);
    }

    return () => {
      resizeObserver?.unobserve(element);
    };
  }, [enabled, memoizedOptions, resizeObserver, target]);
}
```